neurodiffeq/generators.py

```diff
@@ -947,12 +954,17 @@ def __init__(self, generator):
         super(SamplerGenerator, self).__init__()
         self.generator = generator
         self.size = generator.size
+        self._requires_training = generator._requires_training
+        self.examples_history = []
+        if self._requires_training:
+          self.C1 = generator.C1
 
     def get_examples(self) -> List[torch.Tensor]:
         samples = self.generator.get_examples()
         if isinstance(samples, torch.Tensor):
             samples = [samples]
         samples = [u.reshape(-1, 1) for u in samples]
+        self.examples_history.append([sample.detach().numpy() for sample in samples])
```

Comment:
We better call `.cpu()` before calling `.numpy()` in case the tensor is stored on GPU.